pysignalscope/scope.py

```diff
@@ -949,86 +1007,79 @@ def short_to_period(self, f0: Union[float, int, None] = None, time_period: Union
             end_time = start_time + time_period
         elif f0 is not None:
             end_time = start_time + 1/f0
-        self.modify(channel_time_cut_min=start_time, channel_time_cut_max=end_time)
+        channel = HandleScope.modify(channel, channel_time_cut_min=start_time, channel_time_cut_max=end_time)
         # Log flow control
```

Comment:
Overwriting the origin channel data is not allowed. So before this line a copy of the channel data can solve this issue.

Reply:
As modify itself contains a deepcopy now, this should be fine